packages/aws-rfdk/lib/deadline/lib/repository.ts

```diff
@@ -934,6 +1004,12 @@ export class Repository extends Construct implements IRepository {
       '-v', version.linuxFullVersionString(),
     ];
 
+    if (this.secretsManagementSettings.enabled) {
+      installerArgs.push('-r', Stack.of(this.secretsManagementSettings.credentials ?? this).region);
+      this.secretsManagementSettings.credentials?.grantRead(installerGroup);
+      installerArgs.push('-c', this.secretsManagementSettings.credentials?.secretArn ?? '');
```

Comment:
```suggestion
      this.secretsManagementSettings.credentials!.grantRead(installerGroup);
      installerArgs.push('-c', this.secretsManagementSettings.credentials!.secretArn ?? '');
```
At this point in the code we should be certain that the credentials are populated.